src/backend/distributed/utils/metadata_cache.c

```diff
@@ -1184,6 +1190,11 @@ BuildCachedShardList(DistTableCacheEntry *cacheEntry)
 		int placementOffset = 0;
 		int numberOfPlacements = 0;
 
+		/*
+		 * Disable interrupts while making changes for this shard to ensure
+		 * the cache stays in a consistent state.
+		 */
+		HOLD_INTERRUPTS();
```

Comment:
can we actually hit a CHECK_FOR_INTERRUPTS in this code block?

Reply:
This took a while to pinpoint. I had trouble using GDB when interrupts happen, and using `elog()` itself causes additional `CHECK_FOR_INTERRUPTS()`, so I just used [`fprintf()` + `backtrace()`](https://www.gnu.org/software/libc/manual/html_node/Backtraces.html) to find this call path.

`BuildShardPlacementList()` -> `systable_getnext()` -> `index_getnext()` -> `index_getnext_tid()` -> `btgettuple()` -> `_bt_next()` -> `_bt_steppage()` -> `_bt_readnextpage()` -> `CHECK_FOR_INTERRUPTS()`.

Anywhere heap access or index access happens, `CHECK_FOR_INTERRUPTS()` is expected.